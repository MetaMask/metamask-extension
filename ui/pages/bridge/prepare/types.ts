import { InternalAccount } from '@metamask/keyring-internal-api';

export type ExternalAccount = {
  address: string;
  metadata: {
    name: string;
  };
  isExternal: boolean;
};

export type DestinationAccount = InternalAccount | ExternalAccount;
