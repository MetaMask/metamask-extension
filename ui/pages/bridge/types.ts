// Types copied from Metabridge API
export enum BridgeFlag {
  EXTENSION_CONFIG = 'extension-config',
  EXTENSION_SUPPORT = 'extension-support',
  NETWORK_SRC_ALLOWLIST = 'src-network-allowlist',
  NETWORK_DEST_ALLOWLIST = 'dest-network-allowlist',
}

export type FeatureFlagResponse = {
  [BridgeFlag.EXTENSION_CONFIG]: {
    refreshRate: number;
    maxRefreshCount: number;
  };
  [BridgeFlag.EXTENSION_SUPPORT]: boolean;
  [BridgeFlag.NETWORK_SRC_ALLOWLIST]: number[];
  [BridgeFlag.NETWORK_DEST_ALLOWLIST]: number[];
};

export type BridgeAsset = {
  chainId: ChainId;
  address: string;
  symbol: string;
  name: string;
  decimals: number;
  icon?: string;
};

export type QuoteRequest = {
  walletAddress: string;
  destWalletAddress?: string;
  srcChainId: ChainId;
  destChainId: ChainId;
  srcTokenAddress: string;
  destTokenAddress: string;
  srcTokenAmount: string;
  slippage: number;
  aggIds?: string[];
  bridgeIds?: string[];
  insufficientBal?: boolean;
  resetApproval?: boolean;
  refuel?: boolean;
};

type Protocol = {
  name: string;
  displayName?: string;
  icon?: string;
};

enum ActionTypes {
  BRIDGE = 'bridge',
  SWAP = 'swap',
  REFUEL = 'refuel',
}

type Step = {
  action: ActionTypes;
  srcChainId: ChainId;
  destChainId?: ChainId;
  srcAsset: BridgeAsset;
  destAsset: BridgeAsset;
  srcAmount: string;
  destAmount: string;
  protocol: Protocol;
};

type RefuelData = Step;

export type Quote = {
  requestId: string;
  srcChainId: ChainId;
  srcAsset: BridgeAsset;
  srcTokenAmount: string;
  destChainId: ChainId;
  destAsset: BridgeAsset;
  destTokenAmount: string;
  feeData: Record<FeeType.METABRIDGE, FeeData> &
    Partial<Record<FeeType, FeeData>>;
  bridgeId: string;
  bridges: string[];
  steps: Step[];
  refuel?: RefuelData;
};

export type QuoteResponse = {
  quote: Quote;
  approval: TxData | null;
  trade: TxData;
  estimatedProcessingTimeInSeconds: number;
};

enum ChainId {
  ETH = 1,
  OPTIMISM = 10,
  BSC = 56,
  POLYGON = 137,
  ZKSYNC = 324,
  BASE = 8453,
  ARBITRUM = 42161,
  AVALANCHE = 43114,
  LINEA = 59144,
}

export enum FeeType {
  METABRIDGE = 'metabridge',
  REFUEL = 'refuel',
}
export type FeeData = {
  amount: string;
  asset: BridgeAsset;
};
export type TxData = {
  chainId: ChainId;
  to: string;
  from: string;
  value: string;
  data: string;
  gasLimit: number | null;
};
