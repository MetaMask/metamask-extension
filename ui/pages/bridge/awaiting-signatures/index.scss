.awaiting-bridge-signatures {
  display: flex;
  flex-flow: column;
  align-items: center;
  flex: 1;
  width: 100%;
  margin-top: 40%;

  &__content {
    flex-flow: column;
  }

  div {
    text-align: center;
    display: flex;
    justify-content: center;
  }

  &__steps {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    margin: 16px auto 12px auto;

    li {
      margin-bottom: 4px;
      display: flex;
      align-items: center;
    }
  }
}
