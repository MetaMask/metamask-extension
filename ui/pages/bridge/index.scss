@use "design-system";

@import 'prepare/index';
@import 'quotes/index';


.bridge {
  max-height: 100vh;
  width: 360px;
  position: relative;

  &__container {
    width: 100%;

    .multichain-page-footer {
      position: absolute;
      width: 100%;
      height: 80px;
      bottom: 0;
      padding: 16px;
      display: flex;

      button {
        flex: 1;
        height: 100%;
        font-size: 14px;
        font-weight: 500;
      }
    }
  }
}
