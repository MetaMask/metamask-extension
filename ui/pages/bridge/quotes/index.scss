@use "design-system";

.quote-card {
  flex-direction: column;
  display: flex;
  text-align: center;

  p {
    font-size: 12px;
  }


  &__content {
    padding: 16px;
    gap: 4px;
  }

  &__timer {
    height: 32px;

    p {
      color: var(--color-text-alternative);
    }
  }

  .bridge-box > &__footer {
    gap: 0;

    p {
      color: var(--color-text-alternative);
    }
  }

  &__info-row {
    display: flex;
    flex-direction: row;
    justify-content: space-between;

    &__label {
      display: inline-flex;
      gap: 4px;

      p {
        font-weight: var(--font-weight-medium);
        font-size: 14px;
        white-space: nowrap;
      }

      &__tooltip {
        align-items: center;
        height: 100%;
      }
    }

    &__description {
      display: inline-flex;
      gap: 4px;

      &__secondary {
        color: var(--color-text-alternative);
      }
    }
  }
}

.quotes-modal {
  &__column-header {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
  }

  &__quotes {
    display: flex;
    flex-direction: column;

    &__row {
      display: flex;
      flex-direction: row;
      justify-content: space-between;
    }
  }
}
