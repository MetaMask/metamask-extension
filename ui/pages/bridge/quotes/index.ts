export { MultichainBridgeQuoteCard } from './multichain-bridge-quote-card';
export { BridgeQuotesModal } from './bridge-quotes-modal';
