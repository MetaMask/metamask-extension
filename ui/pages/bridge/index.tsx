import React, { useContext, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Switch, useHistory } from 'react-router-dom';
import { I18nContext } from '../../contexts/i18n';
import { clearSwapsState } from '../../ducks/swaps/swaps';
import {
  DEFAULT_ROUTE,
  SWAPS_MAINTENANCE_ROUTE,
  PREPARE_SWAP_ROUTE,
  CROSS_CHAIN_SWAP_ROUTE,
} from '../../helpers/constants/routes';
import { resetBackgroundSwapsState } from '../../store/actions';
import FeatureToggledRoute from '../../helpers/higher-order-components/feature-toggled-route';
import {
  ButtonIcon,
  ButtonIconSize,
  IconName,
} from '../../components/component-library';
import { getIsBridgeChain, getIsBridgeEnabled } from '../../selectors';
import useBridging from '../../hooks/bridge/useBridging';
import {
  Content,
  Footer,
  Header,
} from '../../components/multichain/pages/page';
import { getProviderConfig } from '../../ducks/metamask/metamask';
import { resetInputFields, setFromChain } from '../../ducks/bridge/actions';
import PrepareBridgePage from './prepare/prepare-bridge-page';
import { BridgeCTAButton } from './prepare/bridge-cta-button';

const CrossChainSwap = () => {
  const t = useContext(I18nContext);

  useBridging();

  const history = useHistory();
  const dispatch = useDispatch();

  const isBridgeEnabled = useSelector(getIsBridgeEnabled);
  const providerConfig = useSelector(getProviderConfig);
  const isBridgeChain = useSelector(getIsBridgeChain);

  useEffect(() => {
    isBridgeChain &&
      isBridgeEnabled &&
      providerConfig &&
      dispatch(setFromChain(providerConfig.chainId));

    return () => {
      dispatch(resetInputFields());
    };
  }, [isBridgeChain, isBridgeEnabled, providerConfig]);

  const redirectToDefaultRoute = async () => {
    history.push({
      pathname: DEFAULT_ROUTE,
      state: { stayOnHomePage: true },
    });
    dispatch(clearSwapsState());
    await dispatch(resetBackgroundSwapsState());
  };

  return (
    <div className="bridge">
      <div className="bridge__container">
        <Header
          className="bridge__header"
          startAccessory={
            <ButtonIcon
              iconName={IconName.ArrowLeft}
              size={ButtonIconSize.Sm}
              ariaLabel={t('back')}
              onClick={redirectToDefaultRoute}
            />
          }
          endAccessory={
            <ButtonIcon
              iconName={IconName.Setting}
              size={ButtonIconSize.Sm}
              ariaLabel={t('settings')}
            />
          }
        >
          {t('bridge')}
        </Header>
        <Content className="bridge__content">
          <Switch>
            <FeatureToggledRoute
              redirectRoute={SWAPS_MAINTENANCE_ROUTE}
              flag={isBridgeEnabled}
              path={CROSS_CHAIN_SWAP_ROUTE + PREPARE_SWAP_ROUTE}
              render={() => {
                return <PrepareBridgePage />;
              }}
            />
          </Switch>
        </Content>
        <Footer>
          <BridgeCTAButton />
        </Footer>
      </div>
    </div>
  );
};

export default CrossChainSwap;
