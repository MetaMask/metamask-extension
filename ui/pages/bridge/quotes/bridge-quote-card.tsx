import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import {
  Box,
  Button,
  ButtonVariant,
  Text,
} from '../../../components/component-library';
import {
  getBridgeQuotes,
  getRecommendedQuote,
} from '../../../ducks/bridge/selectors';
import { useI18nContext } from '../../../hooks/useI18nContext';
import { getQuoteDisplayData } from '../utils/quote';
import { useCountdownTimer } from '../../../hooks/bridge/useCountdownTimer';
import MascotBackgroundAnimation from '../../swaps/mascot-background-animation/mascot-background-animation';
import { QuoteInfoRow } from './quote-info-row';
import { BridgeQuotesModal } from './bridge-quotes-modal';

export const BridgeQuoteCard = () => {
  const t = useI18nContext();
  const recommendedQuote = useSelector(getRecommendedQuote);
  const { isLoading } = useSelector(getBridgeQuotes);

  const { etaInMinutes, totalFees, quoteRate } =
    getQuoteDisplayData(recommendedQuote);

  const secondsUntilNextRefresh = useCountdownTimer();

  const [showAllQuotes, setShowAllQuotes] = useState(false);

  if (isLoading && !recommendedQuote) {
    return (
      <Box>
        <MascotBackgroundAnimation />
      </Box>
    );
  }

  return etaInMinutes && totalFees && quoteRate ? (
    <Box className="quote-card">
      <BridgeQuotesModal
        isOpen={showAllQuotes}
        onClose={() => setShowAllQuotes(false)}
      />
      <Box className="bridge-box quote-card__timer">
        {!isLoading && (
          <Text>{t('swapNewQuoteIn', [secondsUntilNextRefresh])}</Text>
        )}
      </Box>

      <Box className="bridge-box prepare-bridge-page__content quote-card__content">
        <QuoteInfoRow
          label={t('estimatedTime')}
          tooltipText={t('bridgeTimingTooltipText')}
          description={t('bridgeTimingMinutes', [etaInMinutes])}
        />
        <QuoteInfoRow label={t('quoteRate')} description={quoteRate} />
        <QuoteInfoRow
          label={t('totalFees')}
          tooltipText={t('bridgeTotalFeesTooltipText')}
          description={totalFees.fiat}
          secondaryDescription={totalFees?.amount}
        />
      </Box>

      <Box className="bridge-box quote-card__footer">
        <span>
          <Text>{t('swapIncludesMMFee', [0.875])}</Text>
          <Button
            variant={ButtonVariant.Link}
            onClick={() => {
              setShowAllQuotes(true);
            }}
          >
            <Text>{t('viewAllQuotes')}</Text>
          </Button>
        </span>
        <Button variant={ButtonVariant.Link}>
          <Text>{t('termsOfService')}</Text>
        </Button>
      </Box>
    </Box>
  ) : null;
};
