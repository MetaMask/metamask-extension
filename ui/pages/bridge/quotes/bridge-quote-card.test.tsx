import React from 'react';
import { renderWithProvider } from '../../../../test/jest';
import configureStore from '../../../store/store';
import { createBridgeMockStore } from '../../../../test/jest/mock-store';
import { CHAIN_IDS } from '../../../../shared/constants/network';
import mockBridgeQuotesErc20Erc20 from '../../../../test/data/bridge/mock-quotes-erc20-erc20.json';
import mockBridgeQuotesNativeErc20 from '../../../../test/data/bridge/mock-quotes-native-erc20.json';
import { RequestStatus } from '../../../../shared/types/bridge';
import { BridgeQuoteCard } from './bridge-quote-card';

describe('BridgeQuoteCard', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render the recommended quote', async () => {
    const mockStore = createBridgeMockStore({
      featureFlagOverrides: {
        extensionConfig: {
          maxRefreshCount: 5,
          refreshRate: 30000,
          chains: {
            [CHAIN_IDS.MAINNET]: { isActiveSrc: true, isActiveDest: false },
            [CHAIN_IDS.OPTIMISM]: { isActiveSrc: true, isActiveDest: true },
          },
        },
      },
      bridgeSliceOverrides: { fromTokenInputValue: 1 },
      bridgeStateOverrides: {
        quoteRequest: { insufficientBal: false },
        quotesRefreshCount: 1,
        quotes: mockBridgeQuotesErc20Erc20,
        getQuotesLastFetched: Date.now(),
        quotesLoadingStatus: RequestStatus.FETCHED,
      },
    });
    const { container } = renderWithProvider(
      <BridgeQuoteCard />,
      configureStore(mockStore),
    );

    expect(container).toMatchSnapshot();
  });

  it('should render the recommended quote while loading new quotes', async () => {
    const mockStore = createBridgeMockStore({
      featureFlagOverrides: {
        extensionConfig: {
          chains: {
            [CHAIN_IDS.MAINNET]: { isActiveSrc: true, isActiveDest: false },
            [CHAIN_IDS.OPTIMISM]: { isActiveSrc: true, isActiveDest: true },
          },
        },
      },
      bridgeSliceOverrides: { fromTokenInputValue: 1 },
      bridgeStateOverrides: {
        quotes: mockBridgeQuotesNativeErc20,
        getQuotesLastFetched: Date.now() - 5000,
        quotesLoadingStatus: RequestStatus.LOADING,
      },
    });
    const { container, queryByText } = renderWithProvider(
      <BridgeQuoteCard />,
      configureStore(mockStore),
    );

    expect(container).toMatchSnapshot();
    expect(queryByText('New quotes in')).not.toBeInTheDocument();
  });

  it('should not render when there is no quote', async () => {
    const mockStore = createBridgeMockStore({
      featureFlagOverrides: {
        extensionConfig: {
          chains: {
            [CHAIN_IDS.MAINNET]: { isActiveSrc: true, isActiveDest: false },
            [CHAIN_IDS.OPTIMISM]: { isActiveSrc: true, isActiveDest: true },
          },
        },
      },
      bridgeSliceOverrides: { fromTokenInputValue: 1 },
      bridgeStateOverrides: {
        quotes: [],
        getQuotesLastFetched: Date.now() - 5000,
        quotesLoadingStatus: RequestStatus.FETCHED,
      },
    });
    const { container } = renderWithProvider(
      <BridgeQuoteCard />,
      configureStore(mockStore),
    );

    expect(container).toMatchSnapshot();
  });

  it('should not render when there is a quote fetch error', async () => {
    const mockStore = createBridgeMockStore({
      featureFlagOverrides: {
        extensionConfig: {
          chains: {
            [CHAIN_IDS.MAINNET]: { isActiveSrc: true, isActiveDest: false },
            [CHAIN_IDS.OPTIMISM]: { isActiveSrc: true, isActiveDest: true },
          },
        },
      },
      bridgeSliceOverrides: { fromTokenInputValue: 1 },
      bridgeStateOverrides: {
        quotes: [],
        getQuotesLastFetched: Date.now() - 5000,
        quotesLoadingStatus: RequestStatus.ERROR,
      },
    });
    const { container } = renderWithProvider(
      <BridgeQuoteCard />,
      configureStore(mockStore),
    );

    expect(container).toMatchSnapshot();
  });
});
