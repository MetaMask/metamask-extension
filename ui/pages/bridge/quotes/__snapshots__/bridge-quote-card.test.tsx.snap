// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BridgeQuoteCard should not render when there is a quote fetch error 1`] = `<div />`;

exports[`BridgeQuoteCard should not render when there is no quote 1`] = `<div />`;

exports[`BridgeQuoteCard should render the recommended quote 1`] = `
<div>
  <div
    class="mm-box mm-container mm-container--max-width-undefined mm-box--display-flex mm-box--gap-3 mm-box--flex-direction-column mm-box--width-full"
  >
    <div
      class="mm-box mm-container mm-container--max-width-undefined mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-nowrap mm-box--justify-content-space-between mm-box--align-items-center"
    >
      <div
        class="mm-box mm-container mm-container--max-width-undefined mm-box--display-flex mm-box--gap-1 mm-box--flex-direction-row mm-box--flex-wrap-nowrap mm-box--justify-content-flex-start mm-box--align-items-center"
        style="white-space: nowrap;"
      >
        <p
          class="mm-box mm-text mm-text--body-lg-medium mm-box--color-text-default"
        >
          Best price
        </p>
        <div
          class="mm-box"
        >
          <div
            class="mm-box mm-box--display-flex"
          >
            <span
              class="mm-box mm-icon mm-icon--size-sm mm-box--display-inline-block mm-box--color-icon-alternative-soft"
              style="mask-image: url('./images/icons/question.svg');"
            />
          </div>
        </div>
      </div>
      <div
        class="mm-box mm-container mm-container--max-width-undefined mm-box--display-flex mm-box--flex-direction-column mm-box--align-items-flex-end mm-box--width-full mm-box--height-full"
      >
        <a
          class="mm-box mm-text mm-text--body-md mm-box--color-primary-default"
        >
          More quotes
        </a>
      </div>
    </div>
    <div
      class="mm-box mm-container mm-container--max-width-undefined mm-box--display-flex mm-box--gap-1 mm-box--flex-direction-column mm-box--width-full"
    >
      <div
        class="mm-box mm-container mm-container--max-width-undefined mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-nowrap mm-box--justify-content-space-between mm-box--align-items-center"
      >
        <p
          class="mm-box mm-text mm-text--body-md-medium mm-box--color-text-alternative-soft"
        >
          Bridging
        </p>
        <div
          class="mm-box mm-container mm-container--max-width-undefined mm-box--display-flex mm-box--gap-1 mm-box--flex-direction-row mm-box--flex-wrap-nowrap mm-box--justify-content-space-between mm-box--align-items-center"
        >
          <div
            class="mm-box mm-text mm-avatar-base mm-avatar-base--size-xs mm-avatar-network mm-text--body-xs mm-text--text-transform-uppercase mm-box--display-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-text-default mm-box--background-color-transparent mm-box--rounded-md mm-box--border-color-background-default mm-box--border-width-1 box--border-style-solid"
          >
            <img
              alt="Ethereum Mainnet logo"
              class="mm-avatar-network__network-image"
              src="./images/optimism.svg"
            />
          </div>
          <p
            class="mm-box mm-text mm-text--body-md mm-box--color-text-default"
            style="white-space: nowrap;"
          >
            OP Mainnet
          </p>
          <span
            class="mm-box mm-icon mm-icon--size-xs mm-box--display-inline-block mm-box--color-inherit"
            style="mask-image: url('./images/icons/arrow-2-right.svg');"
          />
          <div
            class="mm-box mm-text mm-avatar-base mm-avatar-base--size-xs mm-avatar-network mm-text--body-xs mm-text--text-transform-uppercase mm-box--display-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-text-default mm-box--background-color-transparent mm-box--rounded-md mm-box--border-color-background-default mm-box--border-width-1 box--border-style-solid"
          >
            <img
              alt="network logo"
              class="mm-avatar-network__network-image"
              src="./images/pol-token.svg"
            />
          </div>
          <p
            class="mm-box mm-text mm-text--body-md mm-box--color-text-default"
            style="white-space: nowrap;"
          >
            Polygon
          </p>
        </div>
      </div>
      <div
        class="mm-box mm-container mm-container--max-width-undefined mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-nowrap mm-box--justify-content-space-between mm-box--align-items-center"
      >
        <p
          class="mm-box mm-text mm-text--body-md-medium mm-box--color-text-alternative-soft"
        >
          Network fees
        </p>
        <div
          class="mm-box mm-container mm-container--max-width-undefined mm-box--display-flex mm-box--gap-1 mm-box--flex-direction-row mm-box--flex-wrap-nowrap mm-box--justify-content-space-between mm-box--align-items-center"
        >
          <p
            class="mm-box mm-text mm-text--body-md"
          >
            $2.52
            -
            $2.52
          </p>
          <span
            class="mm-box mm-icon mm-icon--size-md mm-box--display-inline-block mm-box--color-icon-alternative-soft"
            style="mask-image: url('./images/icons/swap-vertical.svg'); cursor: pointer;"
          />
        </div>
      </div>
      <div
        class="mm-box mm-container mm-container--max-width-undefined mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-nowrap mm-box--justify-content-space-between mm-box--align-items-center"
      >
        <p
          class="mm-box mm-text mm-text--body-md-medium mm-box--color-text-alternative-soft"
        >
          Time
        </p>
        <p
          class="mm-box mm-text mm-text--body-md mm-box--color-text-default"
        >
          1 min
        </p>
      </div>
      <div
        class="mm-box mm-container mm-container--max-width-undefined mm-box--display-flex mm-box--gap-2 mm-box--flex-direction-row mm-box--flex-wrap-nowrap mm-box--justify-content-flex-start mm-box--align-items-center"
      >
        <p
          class="mm-box mm-text mm-text--body-md mm-box--color-text-alternative-soft"
        >
          Rate includes 0.875% fee
        </p>
        <p
          class="mm-box mm-text mm-text--body-md mm-box--color-text-alternative-soft"
        >
          Â·
        </p>
        <a
          class="mm-box mm-text mm-button-base mm-button-link mm-button-link--size-auto mm-text--body-md mm-box--padding-0 mm-box--padding-right-0 mm-box--padding-left-0 mm-box--display-inline-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-text-alternative-soft mm-box--background-color-transparent"
          href="https://consensys.io/terms-of-use/"
          rel="noopener noreferrer"
          style="text-decoration: underline;"
          target="_blank"
        >
          Terms
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`BridgeQuoteCard should render the recommended quote while loading new quotes 1`] = `
<div>
  <div
    class="mm-box mm-container mm-container--max-width-undefined mm-box--display-flex mm-box--gap-3 mm-box--flex-direction-column mm-box--width-full"
  >
    <div
      class="mm-box mm-container mm-container--max-width-undefined mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-nowrap mm-box--justify-content-space-between mm-box--align-items-center"
    >
      <div
        class="mm-box mm-container mm-container--max-width-undefined mm-box--display-flex mm-box--gap-1 mm-box--flex-direction-row mm-box--flex-wrap-nowrap mm-box--justify-content-flex-start mm-box--align-items-center"
        style="white-space: nowrap;"
      >
        <p
          class="mm-box mm-text mm-text--body-lg-medium mm-box--color-text-default"
        >
          Best price
        </p>
        <div
          class="mm-box"
        >
          <div
            class="mm-box mm-box--display-flex"
          >
            <span
              class="mm-box mm-icon mm-icon--size-sm mm-box--display-inline-block mm-box--color-icon-alternative-soft"
              style="mask-image: url('./images/icons/question.svg');"
            />
          </div>
        </div>
      </div>
      <div
        class="mm-box mm-container mm-container--max-width-undefined mm-box--display-flex mm-box--flex-direction-column mm-box--align-items-flex-end mm-box--width-full mm-box--height-full"
      >
        <a
          class="mm-box mm-text mm-text--body-md mm-box--color-primary-default"
        >
          More quotes
        </a>
      </div>
    </div>
    <div
      class="mm-box mm-container mm-container--max-width-undefined mm-box--display-flex mm-box--gap-1 mm-box--flex-direction-column mm-box--width-full"
    >
      <div
        class="mm-box mm-container mm-container--max-width-undefined mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-nowrap mm-box--justify-content-space-between mm-box--align-items-center"
      >
        <p
          class="mm-box mm-text mm-text--body-md-medium mm-box--color-text-alternative-soft"
        >
          Bridging
        </p>
        <div
          class="mm-box mm-container mm-container--max-width-undefined mm-box--display-flex mm-box--gap-1 mm-box--flex-direction-row mm-box--flex-wrap-nowrap mm-box--justify-content-space-between mm-box--align-items-center"
        >
          <div
            class="mm-box mm-text mm-avatar-base mm-avatar-base--size-xs mm-avatar-network mm-text--body-xs mm-text--text-transform-uppercase mm-box--display-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-text-default mm-box--background-color-transparent mm-box--rounded-md mm-box--border-color-background-default mm-box--border-width-1 box--border-style-solid"
          >
            <img
              alt="Ethereum Mainnet logo"
              class="mm-avatar-network__network-image"
              src="./images/optimism.svg"
            />
          </div>
          <p
            class="mm-box mm-text mm-text--body-md mm-box--color-text-default"
            style="white-space: nowrap;"
          >
            OP Mainnet
          </p>
          <span
            class="mm-box mm-icon mm-icon--size-xs mm-box--display-inline-block mm-box--color-inherit"
            style="mask-image: url('./images/icons/arrow-2-right.svg');"
          />
          <div
            class="mm-box mm-text mm-avatar-base mm-avatar-base--size-xs mm-avatar-network mm-text--body-xs mm-text--text-transform-uppercase mm-box--display-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-text-default mm-box--background-color-transparent mm-box--rounded-md mm-box--border-color-background-default mm-box--border-width-1 box--border-style-solid"
          >
            <img
              alt="network logo"
              class="mm-avatar-network__network-image"
              src="./images/pol-token.svg"
            />
          </div>
          <p
            class="mm-box mm-text mm-text--body-md mm-box--color-text-default"
            style="white-space: nowrap;"
          >
            Polygon
          </p>
        </div>
      </div>
      <div
        class="mm-box mm-container mm-container--max-width-undefined mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-nowrap mm-box--justify-content-space-between mm-box--align-items-center"
      >
        <p
          class="mm-box mm-text mm-text--body-md-medium mm-box--color-text-alternative-soft"
        >
          Network fees
        </p>
        <div
          class="mm-box mm-container mm-container--max-width-undefined mm-box--display-flex mm-box--gap-1 mm-box--flex-direction-row mm-box--flex-wrap-nowrap mm-box--justify-content-space-between mm-box--align-items-center"
        >
          <p
            class="mm-box mm-text mm-text--body-md"
          >
            $2.52
            -
            $2.52
          </p>
          <span
            class="mm-box mm-icon mm-icon--size-md mm-box--display-inline-block mm-box--color-icon-alternative-soft"
            style="mask-image: url('./images/icons/swap-vertical.svg'); cursor: pointer;"
          />
        </div>
      </div>
      <div
        class="mm-box mm-container mm-container--max-width-undefined mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-nowrap mm-box--justify-content-space-between mm-box--align-items-center"
      >
        <p
          class="mm-box mm-text mm-text--body-md-medium mm-box--color-text-alternative-soft"
        >
          Time
        </p>
        <p
          class="mm-box mm-text mm-text--body-md mm-box--color-text-default"
        >
          1 min
        </p>
      </div>
      <div
        class="mm-box mm-container mm-container--max-width-undefined mm-box--display-flex mm-box--gap-2 mm-box--flex-direction-row mm-box--flex-wrap-nowrap mm-box--justify-content-flex-start mm-box--align-items-center"
      >
        <p
          class="mm-box mm-text mm-text--body-md mm-box--color-text-alternative-soft"
        >
          Rate includes 0.875% fee
        </p>
        <p
          class="mm-box mm-text mm-text--body-md mm-box--color-text-alternative-soft"
        >
          Â·
        </p>
        <a
          class="mm-box mm-text mm-button-base mm-button-link mm-button-link--size-auto mm-text--body-md mm-box--padding-0 mm-box--padding-right-0 mm-box--padding-left-0 mm-box--display-inline-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-text-alternative-soft mm-box--background-color-transparent"
          href="https://consensys.io/terms-of-use/"
          rel="noopener noreferrer"
          style="text-decoration: underline;"
          target="_blank"
        >
          Terms
        </a>
      </div>
    </div>
  </div>
</div>
`;
